src/violations.rs

```diff
@@ -0,0 +1,5689 @@
+use itertools::Itertools;
+
+use crate::define_violation;
+use crate::flake8_debugger::types::DebuggerUsingType;
+use crate::flake8_pytest_style::types::{
+    ParametrizeNameType, ParametrizeValuesRowType, ParametrizeValuesType,
+};
+use crate::flake8_quotes::settings::Quote;
+use crate::flake8_tidy_imports::settings::Strictness;
+use crate::pyupgrade::types::Primitive;
+use crate::registry::{
+    Branch, DeferralKeyword, EqCmpop, IsCmpop, LiteralType, MockReference, UnusedCodes,
+};
+use crate::violation::{AlwaysAutofixableViolation, Violation};
+
+// pycodestyle errors
+
+define_violation!(
+    pub struct MultipleImportsOnOneLine;
+);
+impl Violation for MultipleImportsOnOneLine {
+    fn message(&self) -> String {
+        "Multiple imports on one line".to_string()
+    }
+
+    fn placeholder() -> Self {
+        MultipleImportsOnOneLine
+    }
+}
+
+define_violation!(
+    pub struct ModuleImportNotAtTopOfFile;
+);
+impl Violation for ModuleImportNotAtTopOfFile {
+    fn message(&self) -> String {
+        "Module level import not at top of file".to_string()
+    }
+
+    fn placeholder() -> Self {
+        ModuleImportNotAtTopOfFile
+    }
+}
+
+define_violation!(
+    pub struct LineTooLong(pub usize, pub usize);
+);
+impl Violation for LineTooLong {
+    fn message(&self) -> String {
+        let LineTooLong(length, limit) = self;
```

Comment:
Similarly, can this be `let Self(length, limit) = self`?

Reply:
Yes that works. However I would rather have us establish the practice of using named fields instead e.g.

```rust
struct LineTooLong { length: usize, limit: usize }
```

in which case we could get rid of that `let` statement entirely and just use `self.length` and `self.limit`.

(but again I think that's something for future PRs)

Reply:
I agree with this.